packages/web-wallet/src/components/views/CredentialsList/index.tsx

```diff
@@ -258,7 +284,117 @@ const CredentialsList: FC<Props> = (props: Props): ReactElement => {
   if (credentialsLoading || partiesLoading) {
     return <div>{translate('data_provider_loading_message')}</div>
   }
-  return <SSITableView data={credentialTableItems} columns={columns} actions={buildActionList()} onRowClick={onShowCredentialDetails} />
+
+  const onImportCredential = async (file: File): Promise<void> => {
+    const rawCredential = await file.text()
+    const uniformCredential = CredentialMapper.toUniformCredential(rawCredential)
+    const { issuerName, issuerAlias} = getCredentialIssuerNameAndAlias({ verifiableCredential: uniformCredential as VerifiableCredential })
+    const correlationId = CredentialMapper.issuerCorrelationIdFromIssuerType(uniformCredential.issuer)
+    const filter: FindPartyArgs = [
+      {
+        identities: {
+          identifier: {
+            correlationId,
+          },
+        },
+      },
+    ]
+
+    const parties: Array<Party> = await agent.cmGetContacts({
+      filter,
+    })
+
+    if (parties.length === 0) {
+      const contact: AddContactArgs = {
+        legalName: issuerName,
+        displayName: issuerAlias,
+        contactType: {
+          id: '3875c12e-fdaa-4ef6-a340-c936e054b627',
+          origin: PartyOrigin.EXTERNAL,
+          type: PartyTypeType.ORGANIZATION,
+          name: 'Sphereon_default_type',
+          tenantId: '95e09cfc-c974-4174-86aa-7bf1d5251fb4',
+        },
+        identities: [
+          {
+            alias: correlationId,
+            roles: [CredentialRole.ISSUER],
+            origin: IdentityOrigin.EXTERNAL,
+            identifier: {
+              type: CorrelationIdentifierType.DID,
+              correlationId,
+            }
+          }
+        ]
+      }
+
+      await addContact(contact)
+    }
+
+    mutateOne(
+        {
+          resource: DataResource.CREDENTIALS,
+          values: {
+            rawDocument: rawCredential,
+            credentialRole: CredentialRole.HOLDER,
+            credentialId: uniformCredential.id ?? computeEntryHash(rawCredential),
+            issuerCorrelationId: correlationId,
+            issuerCorrelationType: CredentialCorrelationType.DID,
+          },
+        },
+        {
+          onSuccess: () => refetchParties().then(() => onCloseImportCredentialModal())
+        },
+    )
+  }
+
+  const onValidateCredential = async (file: File): Promise<boolean> => {
+    const filePattern = /\.(json|txt)$/i;
+
+    if (!filePattern.test(file.name)) {
+      return false
+    }
+
+    const rawCredential = await file.text()
+    const uniformCredential = CredentialMapper.toUniformCredential(rawCredential)
+
+    const verificationResult = await agent.verifyCredential({
```

Comment:
that is not an agent method. but my idea was to refactor some stuff so that we have these verification methods on a more centralized place when we support sd-jwt. it is kinda heavy if you just want to verify a vc to import the whole holder plugin. maybe we can make a verification package or move it into core or something and then use that also in the holder plugin.

but this has to be discussed.

Reply:
Ok, please copy this comment into a Jira ticket and put a comment there with issue id. I will resolve

Reply:
done